internal/resources/metal/project/resource_test.go

```diff
@@ -211,17 +216,17 @@ func TestAccMetalProject_update(t *testing.T) {
 	})
 }
 
-func testAccCheckMetalSameProject(t *testing.T, before, after *packngo.Project) resource.TestCheckFunc {
+func testAccCheckMetalSameProject(t *testing.T, before, after *metalv1.Project) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		if before.ID != after.ID {
-			t.Fatalf("Expected device to be the same, but it was recreated: %s -> %s", before.ID, after.ID)
+		if before.GetId() != after.GetId() {
+			t.Fatalf("Expected device to be the same, but it was recreated: %s -> %s", before.GetId(), after.GetId())
 		}
 		return nil
 	}
 }
 
 func TestAccMetalProject_BGPUpdate(t *testing.T) {
-	var p1, p2, p3 packngo.Project
+	var p1, p2, p3 metalv1.Project
```

Comment:
This test is failing, which adds weight to avoiding the introduction of new `RequiresReplace` behavior in this PR:
```
=== NAME  TestAccMetalProject_BGPUpdate
    resource_test.go:222: Expected device to be the same, but it was recreated: 5025a781-8ff1-4537-876f-6d12c5552cd7 -> f308175d-fef9-46f8-a461-34698404e38f
```